rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/AdjunctResource.java

```diff
@@ -0,0 +1,264 @@
+        else throw WebResourceUtils.badRequest("%s does not support start/resume", adjunct);
+        
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+
+    @Override
+    public Response stop(String application, String entityToken, String adjunctId) {
+        EntityAdjunct adjunct = brooklyn().getAdjunct(application, entityToken, adjunctId);
+        if (adjunct instanceof Policy) ((Policy)adjunct).suspend();
+        else if (adjunct instanceof Feed) ((Feed)adjunct).suspend();
+        else throw WebResourceUtils.badRequest("%s does not support suspend", adjunct);
+        
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+
+    @Override
+    public Response destroy(String application, String entityToken, String adjunctId) {
+        Entity entity = brooklyn().getEntity(application, entityToken);
+        EntityAdjunct adjunct = brooklyn().getAdjunct(entity, adjunctId);
+
+        if (adjunct instanceof Policy) {
+            ((Policy)adjunct).suspend();
+            entity.policies().remove((Policy) adjunct);
+        } else if (adjunct instanceof Enricher) {
+            entity.enrichers().remove((Enricher) adjunct);
+        } else if (adjunct instanceof Feed) {
+            ((Feed)adjunct).suspend();
+            ((EntityInternal)entity).feeds().remove((Feed) adjunct);
+        } else {
+            throw WebResourceUtils.badRequest("Unexpected adjunct type %s", adjunct);
+        }
+        
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+    
+    // ---- config ----
+    
+    @Override
+    public List<ConfigSummary> listConfig(
+            final String application, final String entityToken, final String adjunctToken) {
+        Entity entity = brooklyn().getEntity(application, entityToken);
+        EntityAdjunct adjunct = brooklyn().getAdjunct(entity, adjunctToken);
+
+        List<ConfigSummary> result = Lists.newArrayList();
+        for (ConfigKey<?> key : adjunct.config().findKeysPresent(Predicates.alwaysTrue())) {
+            result.add(ConfigTransformer.of(key).on(entity, adjunct).includeLinks(ui.getBaseUriBuilder(), false, true).transform());
+        }
+        return result;
+    }
+
+    // TODO support parameters  ?show=value,summary&name=xxx &format={string,json,xml}
+    // (and in sensors class)
+    @Override
+    public Map<String, Object> batchConfigRead(String application, String entityToken, String adjunctToken) {
+        // TODO: add test
+        return EntityTransformer.getConfigValues(brooklyn(), brooklyn().getAdjunct(application, entityToken, adjunctToken) );
+    }
+
+    @Override
+    public String getConfig(String application, String entityToken, String adjunctToken, String configKeyName) {
+        EntityAdjunct adjunct = brooklyn().getAdjunct(application, entityToken, adjunctToken);
+        Set<ConfigKey<?>> cki = adjunct.config().findKeysDeclared(ConfigPredicates.nameSatisfies(Predicates.equalTo(configKeyName)));
```

Comment:
Just noticed that this filter doesn't respect deprecated names - the `ConfigPredicates.nameSatisfies` just uses `key.getName()`.

Should we add another predicate for ConfigPredicates.nameOrDeprecatedNameSatisfies`? Same applies for `setConfig`.

(Not really for this PR, but interested in your thoughts.)

Reply:
given contract of `deprecated`, yes we probably should.  but not quite as you describe as we should prefer non-deprecated config name?